### PR TITLE
Implement Contact Us Form Funtionalities

### LINKED ISSUE #135: Implement Contact Us form functionality
**What do we need to build or fix?**

We need to implement the functionality of the ContactUsForm component rendered on the Contact Us page.
After investigating the best approach on #127 , we decided to use SendGrid to add the functionality.

**Technical details**
- Start your branch from the [feature/contact-us-form-visual-2](https://github.com/Web-Dev-Path/web-dev-path/tree/feature/contact-us-form-visual-2), PR #134 . Once your PR is submitted, please close the #134 with a message explaining that PR is being finalized through yours.
- We're using the email [hello@webdevpath.co](mailto:hello@webdevpath.co)
- If you need to install any package, please do it with the default project package manager, `yarn`
- Please ping @mariana-caldas on Slack to get the SendGrid API key that will be added to the `.env` file under the new variable `SENDGRID_API_KEY`
- Also, please add the new variable placeholder to the .`env-template` file

**Approach suggestions**

https://betterprogramming.pub/how-to-create-a-working-contact-form-in-next-js-459d1fc992ea
https://www.freecodecamp.org/news/how-to-build-a-working-contact-form-with-sendgrid-and-next-js/

**Deadline**

Please keep in mind that once you assign this task to yourself, you'll need to complete it in 10 days.

**Acceptance criteria**

- Test the section and components in many screen sizes, you can use the Inspect tool for that.
- Please test if the new changes added to the components do not affect the other instances.
- Test the feature in many browsers, such as Chrome, Firefox, Edge, and Safari (MAC).
- Update the CHANGELOG.md file.

### PR DESCRIPTION
**Pull request by cherylli**
|                                    Web Dev Path                                    |
| :--------------------------------------------------------------------------------: |
| [**135**](https://github.com/MarianaSouza/web-dev-path/issues/135) |

continues #134 
closes #135 

#### Have you updated the CHANGELOG.md file? If not, please do it.
yes

#### What is this change?
- [x] add functionalities to the contact us  - emailing the form to  hello@webdevpath.co using send grid
- [x] subscribe to the newsletter when the subscribe checkbox is checked - mailchimp
- [x] google recaptcha (v2 invisible)

#### Were there any complications while making this change?
no

#### How did you verify this change?
run locally

#### When should this be merged?
after review

### CODE REVIEW

**Comment by cherylli** [comment]
Some screenshots, mainly to show errors and messages

image
image
image
image
image

**Comment by cherylli** [comment]
Updated response message, now it will only show "submitting..." and "Your message was sent successfully..." the newsletter status and messages will not be shown unless it's an error.

image

File: components/ContactUsForm.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+import contactUsFormStyles from '@/styles/Contact.module.scss';
+import RevealContentContainer from '@/components/containers/RevealContentContainer';
+import SubmitButton from '@/components/buttons/SubmitButton';
+
+export const ContactUsFormSubscribe = ({ setMsg }) => {
+  return (
+    <MailchimpSubscribe
+      url={process.env.NEXT_PUBLIC_MAILCHIMP_URL}
+      render={({ subscribe, status, message }) => {
+        console.info(`MailChimp (contact form): ${status} - ${message}`);
```
Comment (mariana-caldas) [inline components/ContactUsForm.js:15]: This is great, @cherylli ! ⭐

**Review by mariana-caldas — APPROVED** [review]
Wow, such a great work, @cherylli ! Thanks so much for that! 🎉

**Review by klezi10 — APPROVED** [review]

File: components/ContactUsForm.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      ]);
+    }
+
+    if (data.Subscribe) {
+      subscribe({ EMAIL: data.Email });
+    }
+    reset();
+  }
+
+  console.info('Contact Form: these are errors;', errors);
```
Comment (klezi10) [inline components/ContactUsForm.js:93]: I think this can be removed. An old console logs I put in there for the form inputs

**Pull request merged**